modules/codegen/test/src/smithy4s/codegen/internals/SmithyBuildSpec.scala

```diff
@@ -201,4 +201,32 @@ final class SmithyBuildSpec extends munit.FunSuite {
     assertEquals(Set.empty[os.FilePath], actual.sources.toSet)
     assertEquals(Set.empty[SmithyBuildPlugin], actual.plugins.toSet)
   }
+
+  test("generate json with SmithyBuildJson.toJson") {
+    val actual = SmithyBuildJson.toJson(
```

Comment:
**question:** why do we even have two? `SmithyBuildJson.toJson` and `SmithyBuild.writeJson`? @Baccata

Reply:
I think we should remove the `SmithyBuild` one (it's probably a leftover from a merge mistake). 

I think the reason is that we used to rely on the smithy-build pojo, but it was not flexible enough for what we needed to do. So we went straight-to-json I think. Better look at the history though

Reply:
Should I try to take that on in this fix, or does it belong in its own PR? (That seems like it'll touch a lot more of the system.)

Reply:
You can leave it aside for now, no worries 👍

Reply:
https://github.com/disneystreaming/smithy4s/issues/1613